VOLDEV-145 - Implement article header changes and new rail module

onei opened the pull request:
**Awaiting review by VolDevs**

Ticket: https://wikia-inc.atlassian.net/browse/VOLDEV-145

**Changes:**
- Move edit buttons (and other buttons) back above the page header border. This also fixes CONCF-501.
- Move the page count tally to it's own sidebar module, and move the contribute button to said module as well. Special:Videos keeps it's own tally (and no longer has the page share buttons).

**Cleanup includes:**
- Fixed various code issues, e.g. `array()` -> `[]`, `wfMsg*` -> `wfMessage`, whitespace, etc. to conform with the style guides.
- Altered the `PageHeaderIndexAfterActionButtonPrepared` and `PageHeaderIndexExtraButtons` hooks to only modify the intended variables to reduce the potential for subtle, undocumented behaviour/bugs.
- Moved extension specific buttons to hook callbacks in `PageHeader_Index.php` (per a TODO comment), thus adding `SpecialVideosHooks.php` and `WikiaNewFilesHooks.php`.
- Added icons to the user talk archive edit button (seen when Wall is enabled).
- Fixed inconsistencies in `DefaultSettings.php` (simply picked what seemed to be the more predominant method).
- Removed unused sass, e.g. removing styling related to search in the page header.
- Removed use of globals in controller methods in favour of accessing them via `F::app()` (or `$this` if non-static.
- Overhauled logic for `PageHeaderContoller` and it's templates.
- Prevented modification of URLs for the share buttons in the page header for hub wikis (always use database defaults).
- Deleted `PageHeader_EditBox.php` as it was unused (and it's associated method).

**To be fixed before merging:**
- Remove temp code in `PageHeaderController::executeHubs` (sass loading) and `ContributeController::index` (commented out code for testing wikis with high page counts)
- Move the new sidebar module to the correct place in `BodyController::getRailModuleList` (awaiting confirmation of how it's meant to behave in ticket).
- Rebase to fix conflicts (largely related to curatedcontent).

mattdowdell committed:
VOLDEV-145 - Move various buttons back to the pageheader along with additional cleanup

Cleanup includes:

* Remove page count tally from page header, except in Special:Videos per discussion in ticket.

* Moved extension specific buttons from Index template to new hooks files.

* Fixed PageHeaderIndexAfterActionButtonPrepared and PageHeaderIndexExtraButtons hooks to only allow what they should, removing the possibility of subtle, hard-to-trace bugs.

* Added notes to PageHeaderEditPage hook to describe what sort of changes can be made through using it.

* Removed ArticleSaveComplete hook callback as the memcached key hasn't been set in over 3 years, see commit 3b069a25325b31fc942d11f34a7c1b355b77f320.

* Added callback to ArticleViewCustom hook to prevent revision info moving up into the breadcrumbs. See code comments for more details.

* While fixing the above, another bug that caused the 'back to article' links to move into the breadcrumbs on redirects was fixed.

* Change wfMsg* calls to wfMessage with appropriate output methods.

* Fixed security issue in Hubs pageheader that allowed any URL to be inserted into the page share links via local message customisation.

* Changed classes/id and associated sass to conform to the styleguide naming conventions.

* Used more common structure/classes in templates and sass.

* Remove unused styling, e.g. search in title, tally, temp social-icon, etc.

* Hid the page title on hub main pages, as it's hidden in local CSS already.

* Split out and consolidated title formatting logic to another method.

* Move talk button on archived talk pages to the right of the h1 as it is elsewhere.

* Remove unused EditBox method and template file.

* Fixes CONCF-501 as a side-effect of returning buttons to next to the h1

mattdowdell committed:
complete docs, fix minor whitespace issues, add typehinting

mattdowdell committed:
changes per CR

mattdowdell committed:
more CR changes

mattdowdell committed:
fix mainpage title logic, add margin to buttons to prevent regression

mattdowdell committed:
correct comment on pageheadercontroller, cleanup getRailModuleList in BodyController prior to adding the new rail module

mattdowdell committed:
clean up Body controller/template, remove contribute button from wikiheader

mattdowdell committed:
load controller in defaultsettings, clean up defaultsettings (a lot), add styling for contribute module, swap tally and button to the correct way round, add contribute rail module (needs a permanent location, but this works for testing)

mattdowdell committed:
fix styling in VE, remove original padding in contribute button as it looked weird

mattdowdell committed:
merge with dev to fix conflicts

mattdowdell committed:
fix user talk archive edit button

mattdowdell committed:
finish cleanup on BodyController

mattdowdell committed:
CR changes

mattdowdell committed:
fix conflicts with dev, merge in changes to PageHeaderController::getCuratedContentButton

mattdowdell committed:
merge dev into VOLDEV-145

mattdowdell committed:
minor whitespace fixes

mattdowdell committed:
fix conflicts with dev

mattdowdell committed:
fix syntax error

mattdowdell committed:
set defaults for seo test vars so they don't output warnings in debug mode and mess with wikiheader layout

mattdowdell committed:
remove *Index.php file

mattdowdell committed:
and then re-add it because windows doesn't believe in case sensitive file names

Warkot commented on extensions/wikia/WikiaNewFiles/WikiaNewFiles_setup.php:
```diff
@@ -2,29 +2,29 @@
-// Translations
-$wgExtensionMessagesFiles["WikiaNewFiles"] =  "{$dir}/SpecialNewFiles.i18n.php";
+// i18n
+$wgExtensionMessagesFiles["WikiaNewFiles"] =  __DIR__ . '/SpecialNewFiles.i18n.php';
```
Double quotes spoted. :telescope:

Warkot commented on skins/oasis/modules/PageHeaderController.class.php:
```diff
@@ -460,150 +433,106 @@ public function executeEditPage() {
 			$titleMsg = 'oasis-page-header-preview';
-		}
-		else if ( $isShowChanges ) {
+		} else if ( $isShowChanges ) {
```
If you're cleaning up, shouldn't these be converted from `else if` to `elseif`? http://php.net/manual/pl/control-structures.elseif.php

onei replied:
Thanks, I wasn't aware of that one :)

Warkot replied:
_¡De nada!_ :smiley: Some more here: http://stackoverflow.com/questions/3662412/are-elseif-and-else-if-completely-synonymous

mattdowdell committed:
CR changes: double -> single quotes, else if -> elseif

Warkot commented on extensions/wikia/Forum/ForumHooksHelper.class.php:
```diff
@@ -237,20 +237,30 @@ static function onGetUserPermissionsErrors(Title &$title, User &$user, $action,
+				$action = [
+					'class' => '',
+					'text' => wfMessage( 'viewsource' )->escaped(),
+					'href' => $title->getLocalUrl( ['action' => 'edit'] ),
```
When defining an array, put spaces inside the brackets. Don't put spaces there when pointing to an index. Therefore it should be `[ 'action' => 'edit' ]`, but `$button['action']` below is OK. Wikia Guidelines for reference: https://github.com/Wikia/guidelines/blob/master/PHP/CodingConventions.md#array-syntax.

mattdowdell committed:
CR changes: array whitespace padding

mattdowdell committed:
whitespace around parens

Commit:
SEO-12 Use em dash consistently as a SEO separator

Some languages use em dash (instead of regular dash) to separate page
title from wiki title. We're going to using em dash for those languages
instead of regular dash for the " - Wikia" message as well.

Commit:
SEO-12 WikiaMobile to use Category:Page title again

Commit:
SEO-12 Revert an old hack removing the "Category:" prefix in titles

Commit:
SEO-12 Construct the full HTML title in OutputPage::setHTMLTitle

Used to be distributed to a few files including Article.php and
BlogArticle.php. Now you just call the setHTMLTitle method and
it adds the " - Community Name - Wikia" suffix to the end. This
makes nice titles for all pages including Error and Special pages.

Commit:
SEO-12 Make Global Sitemap's title "Global Sitemap"

Commit:
SEO-12 Make Special:Videos title consistent

Commit:
SEO-12 Update to the ?action=create error page title

Commit:
SEO-12 Make WAM page title consistent

Commit:
SEO-98 Roll back to the old robots.txt

It seems there's a lot more URLs indexed since we rolled out the new
robots.txt. We want to verify it was because of the new robots.txt
logic.

Commit:
enable preference service in reston since instances have been deployed to reston

gbenson-wikia committed:
[SERVICES-908] Increases the length of the temporary generated password from 10 characters to 20.

NOTE: This is an emergency fix, so I’m specifically not trying to make
the generated passwords better yet.

ArturKlajnerok committed:
Merge pull request #8742 from Wikia/SERVICES-908

[SERVICES-908] Increases the length of the temporary generated password from 10 characters to 20.

gbenson-wikia committed:
Merge branch 'dev' into SERVICES-908

gbenson-wikia committed:
Correct comment to match code

gbenson-wikia committed:
Merge pull request #8743 from Wikia/SERVICES-908

SERVICES 908

Commit:
Merge pull request #8744 from Wikia/OPS-7272

Video (OPS-7272): Update api url for ustream and viddler

Commit:
Merge pull request #8715 from Wikia/SEO-98-revert-to-old-robots-txt

SEO-98 Roll back to the old robots.txt

nmonterroso committed:
Merge pull request #8741 from Wikia/reston-enable

enable preference service in reston

Commit:
Merge pull request #8700 from Wikia/SEO-12-html-titles-for-category-pages

SEO-12 Fix HTML titles for most of the pages

mattdowdell committed:
Merge branch 'dev' into VOLDEV-145

mattdowdell committed:
Merge branch 'VOLDEV-145' of github.com:WikiaVolDev/app into VOLDEV-145
